### DISCUSSION TITLE
File is Audio file

### DISCUSSION THREAD

**Discussion by Darren2k34** [Zeugma440/atldotnet#102] [Ideas]
Hey, firstly this library is pretty cool :)

I'm currently building an app to help me manage my audio files, and I came across this library to help me get the meta data, which I use to populate my model ( WPF ).

One of the functions in my app is to load all * audio * files in the selected directory, I couldn't really find a simple to implement way on the internet to check if a file is an audio file file or not. One suggestion was to compare the extention, but I didn't really like this approach, another talked about MIDI but I don't know what that is. So.. I began to see if I could find one with your library.

My solution is a static method as follows:
`
public static bool IsFileAudioFile(string path )
        {
            if ( string.IsNullOrWhiteSpace( path ) || !File.Exists(path))
                return false;

            Track audio = new Track( path, true );

            return ( audio?.AudioFormat?.ID ?? -1 ) != -1;
        }
`
Really what I think this method achieves is to check if the file is a supported type ( I think o.o ) and this method seems to work, I honestly don't know if this is a good usecase for AudioFormat thingy I know very little about audio, I just noticed that it returns -1 when i tried loading a text file into it.

Do you have something that achieves the same result already? Or could you please implement something more elegant?
Ideally it would be great to have a method that returns true if the file is an audio file, OR just a method that returns true if the file is a supported file ( audio that can be read etc ), OR both.

Hope this gets implemented into the library. :)

**Comment by Zeugma440** [comment]
Heya,


Thanks for your interest in this library !

You may find inspiration for your use case there : https://github.com/Zeugma440/atldotnet/blob/7868d078c65e71dd0a871ddccdbd76baebd9c684/ATL.benchmark/Information/FileFinder.cs#L107

(` FF_BrowseATLAudioFiles` method).

Let me know if that's what you need.

Cheers,

**Reply by Darren2k34** [reply]
I had a quick look, the "IsFormatSupported" method seems to be what Im looking for, but it's private. Also I don't think I have the benchmark namespace in the nuget I downloaded.

I moved the above mentioned "IsFormatSupported" code into a static helper method, UwU

**Reply by Zeugma440** [reply]
I didn't mean you could call that code directly, but rather copy/paste it in your app. 

`ATL.Benchmark` is an app, it is not a part of the library itself.
